### PR TITLE
Fixed modules using Popen

### PR DESCRIPTION
**Pull request by Shaps**
##### SUMMARY
Fixed modules that were using `subprocess.Popen` to use `module.run_command`. This should address CI errors here #22744
##### ISSUE TYPE
 - Bugfix Pull Request

##### COMPONENT NAME
Multiple modules

##### ANSIBLE VERSION
```
ansible 2.4.0 (fix_shell_out_modules 430cc68800) last updated 2017/05/12 15:32:55 (GMT +000)
  config file =
  configured module search path = [u'/home/ubuntu/.ansible/plugins/modules', u'/usr/share/ansible/plugins/modules']
  ansible python module location = /vagrant/lib/ansible
  executable location = /vagrant/bin/ansible
  python version = 2.7.12 (default, Nov 19 2016, 06:48:10) [GCC 5.4.0 20160609]
```

##### ADDITIONAL INFORMATION
N/A

### CODE REVIEW

**Comment by bjolivot** [comment]
LGTM
shipit

File: lib/ansible/modules/system/gconftool2.py [hunk 1]
```diff
@@ -158,11 +156,7 @@ def call(self, call_type, fail_onerr=True):
                 cmd += "--unset {0}".format(self.key)
 
             # Start external command
-            process = Popen([cmd], stdout=PIPE, stderr=PIPE, shell=True)
-
-            # In either case, we will capture the output
-            out = process.stdout.read()
-            err = process.stderr.read()
+            rc, out, err = self.ansible.run_command(cmd, use_unsafe_shell=True)
```
Comment (Akasurde) [inline lib/ansible/modules/system/gconftool2.py:159]: LGTM

**Review by Akasurde — APPROVED** [review]

File: lib/ansible/modules/database/mysql/mysql_db.py [hunk 2]
```diff
@@ -222,16 +221,12 @@ def db_import(module, host, user, password, db_name, target, all_databases, port
...
-        if p1.returncode != 0:
-            stderr1 = p1.stderr.read()
-            return p1.returncode, '', stderr1
+        rc1, stdout1, stderr1 = module.run_command([comp_prog_path, '-dc', target])
+        if rc1 != 0:
+            return rc1, '', stderr1
         else:
-            return p2.returncode, stdout2, stderr2
+            rc2, stdout2, stderr2 = module.run_command(cmd, data=stdout1, binary_data=True)
+            return rc2, stdout2, stderr2
```
Comment (abadger) [inline lib/ansible/modules/database/mysql/mysql_db.py]: The change to this module is the only one that I question.  I'm leaning towards putting this on the allowed list.  The reason is that the code presently in the module is the equivalent of a UNIX pipe.  The decompressor is able to chunk the data from the file to the database program as it decompresses that portion of the file.  The run_command() version has to store all of the data from the decompressor in memory before passing it to the database program.  So the memory usage can balloon in this case.

**Comment by Shaps** [comment]
cc @abadger

**Comment by abadger** [comment]
Merged to devel

**Pull request merged**